Zia ur Rehman
2024-MSDS-10

# Activity: Explore descriptive statistics

## **Introduction**

Data professionals often use descriptive statistics to understand the data they are working with and provide collaborators with a summary of the relative location of values in the data, as well an information about its spread.

For this activity, you are a member of an analytics team for the United States Environmental Protection Agency (EPA). You are assigned to analyze data on air quality with respect to carbon monoxide, a major air pollutant. The data includes information from more than 200 sites, identified by state, county, city, and local site names. You will use Python functions to gather statistics about air quality, then share insights with stakeholders.

## **Step 1: Imports**


Import the relevant Python libraries `pandas` and `numpy`.

In [45]:
# Import relevant Python libraries.

### YOUR CODE HERE ###

import pandas as pd
import numpy as np

The dataset provided is in the form of a .csv file named `c4_epa_air_quality.csv`. It contains a susbet of data from the U.S. EPA. As shown in this cell, the dataset has been automatically loaded in for you. You do not need to download the .csv file, or provide more code, in order to access the dataset and proceed with this lab. Please continue with this activity by completing the following instructions.

In [25]:
# RUN THIS CELL TO IMPORT YOUR DATA.

### YOUR CODE HERE
df = pd.read_csv("c4_epa_air_quality.csv", index_col = 0)

In [26]:
df

date_local            state_name           county_name      city_name  \
0    2018-01-01               Arizona              Maricopa        Buckeye   
1    2018-01-01                  Ohio               Belmont      Shadyside   
2    2018-01-01               Wyoming                 Teton  Not in a city   
3    2018-01-01          Pennsylvania          Philadelphia   Philadelphia   
4    2018-01-01                  Iowa                  Polk     Des Moines   
..          ...                   ...                   ...            ...   
255  2018-01-01  District Of Columbia  District of Columbia     Washington   
256  2018-01-01             Wisconsin                 Dodge       Kekoskee   
257  2018-01-01              Kentucky             Jefferson     Louisville   
258  2018-01-01              Nebraska               Douglas          Omaha   
259  2018-01-01        North Carolina                  Wake  Not in a city   

                                       local_site_name   parameter_name  \
0                                              BUCKEYE  Carbon monoxide   
1                                            Shadyside  Carbon monoxide   
2    Yellowstone National Park - Old Faithful Snow ...  Carbon monoxide   
3                               North East Waste (NEW)  Carbon monoxide   
4                                            CARPENTER  Carbon monoxide   
..                                                 ...              ...   
255                                          Near Road  Carbon monoxide   
256                              HORICON WILDLIFE AREA  Carbon monoxide   
257                                       CANNONS LANE  Carbon monoxide   
258                                                NaN  Carbon monoxide   
259                                         Triple Oak  Carbon monoxide   

      units_of_measure  arithmetic_mean  aqi  
0    Parts per million         0.473684    7  
1    Parts per million         0.263158    5  
2    Parts per million         0.111111    2  
3    Parts per million         0.300000    3  
4    Parts per million         0.215789    3  
..                 ...              ...  ...  
255  Parts per million         0.244444    3  
256  Parts per million         0.200000    2  
257  Parts per million         0.163158    2  
258  Parts per million         0.421053    9  
259  Parts per million         0.188889    2  

[260 rows x 9 columns]

<details>
  <summary><h4><strong>Hint</strong></h4></summary>

  Use the `read_csv` function from the pandas `library`. The `index_col` parameter can be set to `0` to read in the first column as an index (and to avoid `"Unnamed: 0"` appearing as a column in the resulting DataFrame).

</details>

## **Step 2: Data exploration**

To understand how the dataset is structured, display the first 10 rows of the data.

In [27]:
# Display first 10 rows of the data.

### YOUR CODE HERE
df.head(10)

date_local    state_name   county_name      city_name  \
0  2018-01-01       Arizona      Maricopa        Buckeye   
1  2018-01-01          Ohio       Belmont      Shadyside   
2  2018-01-01       Wyoming         Teton  Not in a city   
3  2018-01-01  Pennsylvania  Philadelphia   Philadelphia   
4  2018-01-01          Iowa          Polk     Des Moines   
5  2018-01-01        Hawaii      Honolulu  Not in a city   
6  2018-01-01        Hawaii      Honolulu  Not in a city   
7  2018-01-01  Pennsylvania          Erie           Erie   
8  2018-01-01        Hawaii      Honolulu       Honolulu   
9  2018-01-01      Colorado       Larimer   Fort Collins   

                                     local_site_name   parameter_name  \
0                                            BUCKEYE  Carbon monoxide   
1                                          Shadyside  Carbon monoxide   
2  Yellowstone National Park - Old Faithful Snow ...  Carbon monoxide   
3                             North East Waste (NEW)  Carbon monoxide   
4                                          CARPENTER  Carbon monoxide   
5                                            Kapolei  Carbon monoxide   
6                                            Kapolei  Carbon monoxide   
7                                                NaN  Carbon monoxide   
8                                           Honolulu  Carbon monoxide   
9                      Fort Collins - CSU - S. Mason  Carbon monoxide   

    units_of_measure  arithmetic_mean  aqi  
0  Parts per million         0.473684    7  
1  Parts per million         0.263158    5  
2  Parts per million         0.111111    2  
3  Parts per million         0.300000    3  
4  Parts per million         0.215789    3  
5  Parts per million         0.994737   14  
6  Parts per million         0.200000    2  
7  Parts per million         0.200000    2  
8  Parts per million         0.400000    5  
9  Parts per million         0.300000    6

<details>
  <summary><h4><strong>Hint</strong></h4></summary>

  Use the `head()` function from the `pandas` library.

</details>

**Question:** What does the `aqi` column represent?


[Write your response here. Double-click (or enter) to edit.]
AQI  is Air Quality index a tool that measures, reports daily air quality
What it does
The AQI is a scale that measures air quality from 0 to 500, with 0 being perfect air and 500 being an immediate danger to public health. The AQI also indicates the health effects that may occur within hours or days of breathing polluted air. 
How it works
The AQI is based on measurements of pollutants like ground-level ozone, sulfur dioxide, nitrogen dioxide, carbon monoxide, and fine particles. The AQI breaks air pollution levels into six categories, each with a color and advice. 
What it means
AQI values below 100 are considered satisfactory, while values above 100 indicate unhealthy air quality. The higher the AQI value, the greater the risk of health harm. 
Who developed it
The U.S. Environmental Protection Agency (EPA) developed the AQI to provide a simple, consistent way to report air quality

Now, get a table that contains some descriptive statistics about the data.

In [30]:
# Get descriptive stats.

### YOUR CODE HERE
df.describe()

arithmetic_mean         aqi
count       260.000000  260.000000
mean          0.403169    6.757692
std           0.317902    7.061707
min           0.000000    0.000000
25%           0.200000    2.000000
50%           0.276315    5.000000
75%           0.516009    9.000000
max           1.921053   50.000000

<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `pandas` library that allows you to generate a table of basic descriptive statistics about the numeric columns in a DataFrame.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

  Use the `describe()` function from the `pandas` library.

</details>

**Question:** Based on the table of descriptive statistics, what do you notice about the count value for the `aqi` column?

There are 260 total number of values and the range of data form 0 to 50


In [33]:
percentile_25 = df["aqi"].quantile(0.25)
print(percentile_25)

2.0


**Question:** What do you notice about the 25th percentile for the `aqi` column?

This is an important measure for understanding where the aqi values lie.

only 2 values are ranked as 25 percential it means that 25% of the data points are less than or equal to 2, and the remaining 75% of the data points are greater than 2.

**Question:** What do you notice about the 75th percentile for the `aqi` column?

This is another important measure for understanding where the aqi values lie.

In [36]:
percentile_75 = df["aqi"].quantile(0.75)
print(percentile_75)

9.0


75th percentile is 9, it means that 75% of the data points are less than or equal to 9, and the remaining 25% of the data points are greater than 9

## **Step 3: Statistical tests**

Next, get some descriptive statistics about the states in the data.

In [49]:
# Get descriptive stats about the states in the data.

### YOUR CODE HERE
state_stats = df.groupby('state_name')['aqi'].describe()
print(state_stats)

                      count       mean        std  min    25%   50%    75%  \
state_name                                                                   
Alabama                 1.0   2.000000        NaN  2.0   2.00   2.0   2.00   
Alaska                  2.0   8.500000   0.707107  8.0   8.25   8.5   8.75   
Arizona                14.0  15.214286  14.104983  2.0   5.50  10.0  19.75   
Arkansas                1.0   3.000000        NaN  3.0   3.00   3.0   3.00   
California             66.0  12.121212   7.301244  1.0   7.00  11.0  16.00   
Colorado                9.0   5.000000   1.322876  3.0   5.00   5.0   6.00   
Connecticut             4.0   3.500000   1.914854  1.0   2.50   4.0   5.00   
Delaware                1.0   3.000000        NaN  3.0   3.00   3.0   3.00   
District Of Columbia    2.0   2.500000   0.707107  2.0   2.25   2.5   2.75   
Florida                12.0   5.500000   2.430862  1.0   4.50   6.0   7.00   
Georgia                 2.0   4.000000   2.828427  2.0   3.00   

<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `pandas` library that allows you to generate basic descriptive statistics about a DataFrame or a column you are interested in.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

 Use the `describe()` function from the `pandas` library. Note that this function can be used:
- "on a DataFrame (to find descriptive statistics about the numeric columns)"
- "directly on a column containing categorical data (to find pertinent descriptive statistics)"

</details>

**Question:** What do you notice while reviewing the descriptive statistics about the states in the data?

Note: Sometimes you have to individually calculate statistics. To review to that approach, use the `numpy` library to calculate each of the main statistics in the preceding table for the `aqi` column.

The descriptive statistics provide valuable insights into air quality across different states, highlighting disparities and trends. Understanding these variations is critical for formulating targeted strategies to improve air quality and protect public health.

## **Step 4. Results and evaluation**

Now, compute the mean value from the `aqi` column.

In [39]:
# Compute the mean value from the aqi column.

### YOUR CODE HERE
aqi_mean=df["aqi"].mean()
print(aqi_mean)

6.757692307692308


<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `numpy` library that allows you to get the mean value from an array or a Series of values.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

  Use the `mean()` function from the `numpy` library.

</details>

**Question:** What do you notice about the mean value from the `aqi` column?

This is an important measure, as it tells you what the average air quality is based on the data.

An AQI value of 6.76 falls well within the "Good" air quality range, which typically covers AQI values from 0 to 50, indicating that the air is considered clean and healthy for the general population.

Next, compute the median value from the aqi column.

In [40]:
# Compute the median value from the aqi column.

### YOUR CODE HERE
aqi_median=df["aqi"].median()
print(aqi_median)

5.0


<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `numpy` library that allows you to get the median value from an array or a series of values.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

  Use the `median()` function from the `numpy` library.

</details>

**Question:** What do you notice about the median value from the `aqi` column?

This is an important measure for understanding the central location of the data.

Central Location: A median of 5.0 indicates that half of the AQI readings in your dataset are below 5, and half are above. This value represents the middle of your AQI data.

Comparison with Mean:

If the mean AQI is 6.76 (as you mentioned earlier), the median being lower than the mean suggests that the data might be right-skewed. This means that there are some higher AQI values pulling the mean up, while the bulk of the data is clustered around the lower values.

Next, identify the minimum value from the `aqi` column.

In [41]:
# Identify the minimum value from the aqi column.

### YOUR CODE HERE
aqi_min=df["aqi"].min()
print(aqi_min)

0


<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `numpy` library that allows you to get the minimum value from an array or a Series of values.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

  Use the `min()` function from the `numpy` library.

</details>

**Question:** What do you notice about the minimum value from the `aqi` column?

This is an important measure, as it tell you the best air quality observed in the data.

an AQI minimum value of 0.0 indicates that the air quality has reached optimal conditions at some point in the dataset. This is an important measure for assessing overall air quality trends and potential health impacts.

Now, identify the maximum value from the `aqi` column.

In [42]:
# Identify the maximum value from the aqi column.

### YOUR CODE HERE
api_max=df["aqi"].max()
print(api_max)

50


<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `numpy` library that allows you to get the maximum value from an array or a Series of values.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

  Use the `max()` function from the `numpy` library.

</details>

**Question:** What do you notice about the maximum value from the `aqi` column?

This is an important measure, as it tells you which value in the data corresponds to the worst air quality observed in the data.

an AQI maximum value of 50 indicates that while air quality remains good overall, it is important to monitor these values closely, as they are near the threshold that could lead to moderate air quality levels. This understanding is vital for public health considerations and environmental management.

Now, compute the standard deviation for the `aqi` column.

By default, the `numpy` library uses 0 as the Delta Degrees of Freedom, while `pandas` library uses 1. To get the same value for standard deviation using either library, specify the `ddof` parameter to 1 when calculating standard deviation.

In [44]:
# Compute the standard deviation for the aqi column.

### YOUR CODE HERE
api_std=df['aqi'].std()
print(api_std)

7.061706678820724


<details>
  <summary><h4><strong>Hint 1</strong></h4></summary>

Refer to the video section about descriptive statistics in Python.

</details>

<details>
  <summary><h4><strong>Hint 2</strong></h4></summary>

  There is a function in the `numpy` library that allows you to get the standard deviation from an array or a series of values.

</details>

<details>
  <summary><h4><strong>Hint 3</strong></h4></summary>

Use the `std()` function from the `numpy` library. Make sure to specify the `ddof` parameter as 1. To read more about this function,  refer to its documentation in the references section of this lab.

</details>

**Question:** What do you notice about the standard deviation for the `aqi` column?

This is an important measure of how spread out the aqi values are.

Given that the mean AQI is around 6.76 (from your previous calculations), the standard deviation indicates that many AQI readings are likely distributed within a range of approximately:
Mean ± Standard Deviation: 
6.76 ± 7.06
This suggests that the AQI values may vary widely, possibly ranging from negative values (not applicable in the context of AQI, as values can't be negative) up to higher levels depending on the specific observations in your dataset.

## **Considerations**


**What are some key takeaways that you learned during this lab?**

lab highlighted the significance of descriptive statistics in understanding AQI data, the implications for public health, and the necessity for ongoing monitoring and assessment of air quality to protect community health and well-being

**How would you present your findings from this lab to others? Consider the following relevant points noted by AirNow.gov as you respond:**
- "AQI values at or below 100 are generally thought of as satisfactory. When AQI values are above 100, air quality is considered to be unhealthy—at first for certain sensitive groups of people, then for everyone as AQI values increase."
- "An AQI of 100 for carbon monoxide corresponds to a level of 9.4 parts per million."

I analyzed Air Quality Index (AQI) data, revealing a mean AQI of 6.76 and a median of 5.0, indicating generally good air quality, with a minimum of 0.0 and a maximum of 50. The standard deviation of 7.0617 suggests moderate variability in AQI values. Given that AQI values at or below 100 are considered satisfactory, our data predominantly falls within this range, minimizing health risks. However, as AQI values exceed 100, air quality can become unhealthy, particularly for sensitive groups. For instance, an AQI of 100 for carbon monoxide corresponds to a level of 9.4 parts per million, highlighting the importance of continuous monitoring and intervention strategies to maintain healthy air quality for the community.

**What summary would you provide to stakeholders? Use the same information provided previously from AirNow.gov as you respond.**

AQI values at or below 100 are considered satisfactory, the data primarily falls within this safe range, minimizing health risks. However, as AQI exceeds 100, air quality can become unhealthy, especially for sensitive groups. For example, an AQI of 100 for carbon monoxide corresponds to 9.4 parts per million. Continuous monitoring and proactive strategies are essential to maintain healthy air quality for the community.

**References**

[Air Quality Index - A Guide to Air Quality and Your Health](https://www.airnow.gov/sites/default/files/2018-04/aqi_brochure_02_14_0.pdf). (2014,February)

[Numpy.Std — NumPy v1.23 Manual](https://numpy.org/doc/stable/reference/generated/numpy.std.html)

US EPA, OAR. (2014, 8 July).[*Air Data: Air Quality Data Collected at Outdoor Monitors Across the US*](https://www.epa.gov/outdoor-air-quality-data).

**Congratulations!** You've completed this lab. However, you may not notice a green check mark next to this item on Coursera's platform. Please continue your progress regardless of the check mark. Just click on the "save" icon at the top of this notebook to ensure your work has been logged.